IPython/core/magics/basic.py

```diff
@@ -629,6 +630,9 @@ def notebook(self, s):
                 },
             },
         )
+        # Sign the notebook to make it trusted
+        notary = NotebookNotary()
```

Comment:
```suggestion
        notary = NotebookNotary()
        notary.update_config(shell.config)
```

Without the above change it would use the default options for traitlets like `db_file`, `data_dir`, or `algorithm`.

In the test can you invoke `%config NotebookNotary.algorithm = "sha384"` and then in test verification use `NotebookNotary(algorithm="sha384")`.

<img width="589" height="348" alt="image" src="https://github.com/user-attachments/assets/fea577ec-f422-4792-96bd-e7d4b20422ef" />